### PR TITLE
Introduction to MetaMask API

### PR DESCRIPTION
**Pull request by httpJunkie**
This is the first draft of an article that loosely follows the Introduction to the MetaMask API talk that I presented at SmartCon.

Formatting should not be a huge concern at this point; looking for feedback on the explanation and code.

### CODE REVIEW

File: introduction-to-metamask-api/index.md [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+  - [MetaMask API Playground](#metamask-api-playground)  
+  - [Chainlist](#chainlist)
+  - [Eserialize](#eserialize)
+- [Examples of Usage](#examples-of-usage)  
+- [Connecting to MetaMask Mobile with SDK](#connecting-to-metamask-mobile-with-sdk)
+- [Smart Contract Example](#smart-contract-example)  
+
+## What is the MetaMask API?
+
+Whenever we use a browser like Chrome, Firefox, or Brave and a wallet extension like MetaMask installed, we can use the Ethereum provider (as specified by EIP-1193) injected into the browser page at `window.ethereum`.
```
Comment (GuiBibeau) [inline introduction-to-metamask-api/index.md]: Maybe a link to the EIP?

File: introduction-to-metamask-api/index.md [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+
+We use this provider with our dapp to request users’ Ethereum accounts, read on-chain data and have the user sign messages and transactions.
+
+[Should we have a link to a docs page that describes what happens when multiple wallets are installed?]
+
+We can tell in the developer tools when MetaMask is installed and enabled through the developer tools, using chrome you can look into the Sources/Page, and you will see an icon that represents MetaMask.
+
+![](./images/01-sources-page.png)
```
Comment (GuiBibeau) [inline introduction-to-metamask-api/index.md]: let's add an alt description for accessibility

File: introduction-to-metamask-api/index.md [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+Page is used for viewing available resources on the current webpage. 
+Under the Top level, we see the representation of the main document and all of its resources.
+
+If we see MetaMask represented by a cloud icon here, we know that the MetaMask is installed and not disabled.
+
+The `window.ethereum` object is being injected into the page, allowing us to interact with MetaMask and Ethereum.
+
+We can directly access and call the `window.ethereum` object in our developer tools console. Let’s say that we wanted to get the current chainId…
+
+![](./images/02-console-eth-chainid.png)
```
Comment (GuiBibeau) [inline introduction-to-metamask-api/index.md]: alt description here too

File: introduction-to-metamask-api/index.md [hunk 4]
```diff
@@ -0,0 +1,389 @@
...
+
+By running the code above in our console, and if we are currently connected to Polygon, we would see ‘0x89’ returned from this method call.
+
+As you get started with using MetaMask API methods, you might want to check out some of our [beginner-level videos](https://www.youtube.com/watch?v=03lbmYrawV8) from our Lead DX Engineer [Gui Bibeau](https://twitter.com/guibibeau) that can help get you started working with the MetaMask API.
+
+## What is JSON RPC?
```
Comment (GuiBibeau) [inline introduction-to-metamask-api/index.md:48]: I would maybe introduce the relation ship between JSON RPC and MM before describing what it is. Maybe a single sentance in the API section above:

MetaMask API calls are defined with the JSON-RPC specs that we explain further down here [link to the explainer]

File: introduction-to-metamask-api/index.md [hunk 5]
```diff
@@ -0,0 +1,389 @@
...
+
+- [Chainlist](https://chainlist.org "Find Ethereum Networks and Chains and add to MetaMask")
+
+Below is an example using Chainlist, I can see Fantom Opera (mainnet), it's chainId and I can add to MetaMask and in the confirmation see the information used to add the network to my wallet.
+
+![](./images//15-chainlist.png)
+
+### Eserialize
+
+- [eserialize](https://eserialize.com/ "Serialize and deserialize the Ethereum Stack")
```
Comment (GuiBibeau) [inline introduction-to-metamask-api/index.md:209]: `eserialize` => `deserialize` ?

**Review by GuiBibeau — COMMENTED** [review]
I really love this writeout! I have just 1-2 places of concern:

- some pictures need alt descriptions
- JSON RPC could be introduce earlier on before a H2 just to help build into the next section

File: introduction-to-metamask-api/index.md (hunk 5 above)
Comment (httpJunkie) [reply introduction-to-metamask-api/index.md:209]: Eserialize is just the name of the webtool, is there anything I am missing?

File: introduction-to-metamask-api/index.md (hunk 2 above)
Comment (httpJunkie) [reply introduction-to-metamask-api/index.md]: I will ensure that all images have a proper alt description.

File: introduction-to-metamask-api/index.md (hunk 5 above)
Comment (GuiBibeau) [reply introduction-to-metamask-api/index.md:209]: Read quickly and thought this was concerning deserialization.

**Review by GuiBibeau — APPROVED** [review]

**Pull request merged**